nototools/coverage.py

```diff
@@ -18,62 +18,117 @@
-    cmaps = {}
-    for table in cmap_table.tables:
-        if (table.format, table.platformID, table.platEncID) in [
-            (4, 3, 1), (12, 3, 10)]:
-            cmaps[table.format] = table.cmap
-    if 12 in cmaps:
-        cmap = cmaps[12]
-    elif 4 in cmaps:
-        cmap = cmaps[4]
-    else:
-        cmap = {}
-    return frozenset(cmap.keys())
+  """Returns the character coverage of a font.
+
+  Args:
+    font: The input font's file name, or a TTFont.
+
+  Returns:
+    A frozenset listing the characters supported in the font.
+  """
+  if type(font) is str:
+    font = ttLib.TTFont(font)
+  cmap_table = font["cmap"]
+  cmaps = {}
+  for table in cmap_table.tables:
+    if (table.format, table.platformID, table.platEncID) in [
+        (4, 3, 1), (12, 3, 10)]:
+      cmaps[table.format] = table.cmap
+  if 12 in cmaps:
+    cmap = cmaps[12]
+  elif 4 in cmaps:
+    cmap = cmaps[4]
+  else:
+    cmap = {}
+  return frozenset(cmap.keys())
 
 
 def convert_set_to_ranges(charset):
-    """Converts a set of characters to a list of ranges."""
-    working_set = set(charset)
-    output_list = []
-    while working_set:
-        start = min(working_set)
-        end = start + 1
-        while end in working_set:
-            end += 1
-        output_list.append((start, end - 1))
-        working_set.difference_update(range(start, end))
-    return output_list
-
-
-def main(argv):
-    """Outputs the character coverage of fonts given on the command line."""
-    import unicode_data
-    for font in argv[1:]:
-        print "%s:" % font
-        for char in sorted(character_set(font)):
-            try:
-                name = unicode_data.name(char)
-            except ValueError:
-                name = "<Unassigned>"
-            print "U+%04X %s" % (char, name)
+  """Converts a set of characters to a list of ranges."""
+  working_set = set(charset)
+  output_list = []
+  while working_set:
+      start = min(working_set)
+      end = start + 1
+      while end in working_set:
+          end += 1
+      output_list.append((start, end - 1))
+      working_set.difference_update(range(start, end))
+  return output_list
+
+
+def _print_char_info(chars):
+  for char in chars:
+    try:
+      name = unicode_data.name(char)
+    except ValueError:
+      name = "<Unassigned>"
+    print "U+%04X %s" % (char, name)
+
+
+def _write_char_text(chars, filepath, chars_per_line):
+  text = [unichr(cp) for cp in chars if not unicode_data.category(cp)[0] in ['M','C','Z']]
+  filename, _ = path.splitext(path.basename(filepath))
+  m = re.match(r'(.*)-Regular|Bold|Italic|BoldItalic', filename)
+  if m:
+    filename = m.group(1)
+  filename += '_chars.txt'
+  print 'writing file: %s' % filename
+  print '%d characters (of %d)' % (len(text), len(chars))
+  if chars_per_line > 0:
+    lines = []
+    for n in range(0, len(text), chars_per_line):
+      substr = text[n:n+chars_per_line]
+      lines.append(' '.join([cp for cp in substr]))
```

Comment:
Also don't need to actually generate a list here. Might be more efficient in general to just pass in the generator? Not that it really matters anyhow.